modules-js/react-fleet/src/components/DetailsDisclosure.tsx

```diff
@@ -25,42 +35,46 @@ export default function ApostilleRequestInstructions(): JSX.Element {
     const testElement = document.createElement('details');
 
     setDetailsElementIsSupported('open' in testElement);
-    // Because we only need to test for support once, ensure the effect
-    // does not fire again.
-  }, [detailsElementIsSupported]);
+  }, []); // effect only needs to run once
 
   if (detailsElementIsSupported) {
     return (
       <details css={DETAILS_STYLING}>
-        <summary>{summaryContent()}</summary>
+        <summary>{summaryElement(props.summaryContent)}</summary>
 
-        {detailsContent()}
+        {props.children}
       </details>
     );
   } else {
+    // If an id is not passed in, use timestamp to ensure unique ids in case
+    // more than one <DetailsDisclosure> component is present.
+    const id = props.id || Date.now().toString();
```

Comment:
Timestamp isn’t safe when doing SSR -> client rendering

Look for usages of `string-hash` to see how other form elements have generated a stable ID.